README.md

```diff
@@ -233,49 +233,54 @@ Each CUDA version only supports one particular XCode version. The following comb
 
 On Windows
 
-At least Visual Studio 2017 version 15.6 with the toolset 14.13 and [NVTX](https://docs.nvidia.com/gameworks/content/gameworkslibrary/nvtx/nvidia_tools_extension_library_nvtx.htm) are needed.
+Build with CPU
 
-If the version of Visual Studio 2017 is higher than 15.6, installing of "VC++ 2017 version 15.6 v14.13 toolset" is strongly recommended.
-<br/> If the version of Visual Studio 2017 is lesser than 15.6, please update Visual Studio 2017 to the latest version along with installing "VC++ 2017 version 15.6 v14.13 toolset".
-<br/> There is no guarantee of the correct building with VC++ 2017 toolsets, others than version 15.6 v14.13.
-<br/> "VC++ 2017 version 15.6 v14.13 toolset" might be installed onto already installed Visual Studio 2017 by running its installation once again and checking the corresponding checkbox under "Individual components"/"Compilers, build tools, and runtimes".
+It's faily easy to build with CPU. It is recommended to build the latest baseline with Visual Studio 2019 version 16.7.6 or higher. 
 
+Build with Cuda
+
+[NVTX](https://docs.nvidia.com/gameworks/content/gameworkslibrary/nvtx/nvidia_tools_extension_library_nvtx.htm) are needed to build Pytorch with Cuda
 NVTX is a part of CUDA distributive, where it is called "Nsight Compute". To install it onto already installed CUDA run CUDA installation once again and check the corresponding checkbox.
-Be sure that CUDA with Nsight Compute is installed after Visual Studio 2017.
+Be sure that CUDA with Nsight Compute is installed after Visual Studio.
+
+Currently, VS 2017 / 2019, and Ninja are supported as the generator of CMake. If `ninja.exe` is detected in `PATH`, then Ninja will be used as the default generator, otherwise, it will use VS 2017 / 2019.
+<br/> If Ninja is selected as the generator, the latest MSVC wil get selected as the underlying toolchain.
 
-Currently, VS 2017, VS 2019, and Ninja are supported as the generator of CMake. If `ninja.exe` is detected in `PATH`, then Ninja will be used as the default generator, otherwise, it will use VS 2017.
-<br/> If Ninja is selected as the generator, the latest MSVC which is newer than VS 2015 (14.0) will get selected as the underlying toolchain. If you use CMake <= 3.14.2 and has VS 2019 installed, then even if you specify VS 2017 as the generator, VS 2019 will get selected as the generator.
+CUDA and MSVC, MSVC and Pytorch have version dependency, please install the matched versions. 
+| CUDA version | Newest supported VS version                             | Pytorch version |
+| ------------ | ------------------------------------------------------- | --------------- |
+| 9.2          | Visual Studio 2017 Update 5 (15.5) (`_MSC_VER` <= 1912) |       <= 1.6    |
+| 10.0         | Visual Studio 2017 (15.X) (`_MSC_VER` < 1920)           |       <= 1.6    |
+| 10.1 ~ 11.0  | Visual Studio 2019 (16.X) (`_MSC_VER` < 1930)           |       >= 1.7    |
```

Comment:
Is there a suggested python version for this matrix as well?

Reply:
The minimum supported version is 3.6.2, according to setup.py. I added a series of recommended Python version in the Linux guide in #47593 to fix an issue related to anaconda.

Reply:
since the suggestion python version is mentioned at the beginning of https://github.com/pytorch/pytorch#from-source and it's platform agnostic. I think we might not add it once more.